OrbitGl/App.cpp

```diff
@@ -839,32 +832,71 @@ void OrbitApp::SendErrorToUi(const std::string& title,
 }
 
 //-----------------------------------------------------------------------------
-void OrbitApp::EnqueueModuleToLoad(const std::shared_ptr<Module>& a_Module) {
-  m_ModulesToLoad.push_back(a_Module);
+void OrbitApp::EnqueueModuleToLoad(const std::shared_ptr<Module>& module) {
+  if (!modules_currently_loading_.contains(module)) {
+    modules_to_load_.insert(module);
+  }
 }
 
 //-----------------------------------------------------------------------------
 void OrbitApp::LoadModules() {
-  if (!m_ModulesToLoad.empty()) {
-    LoadRemoteModules();
-  }
+  LoadRemoteModules(Capture::GTargetProcess->GetID(), nullptr);
 }
 
 //-----------------------------------------------------------------------------
-void OrbitApp::LoadRemoteModules() {
-  GetSymbolsClient()->LoadSymbolsFromModules(Capture::GTargetProcess.get(),
-                                             m_ModulesToLoad, nullptr);
-  m_ModulesToLoad.clear();
-  // This is a bit counterintuitive. LoadSymbols generates request to
-  // the service if symbols cannot be loaded locally. In which case
-  // the UI update happens in OnRemoteModuleDebugInfo. But if all symbols
-  // are loaded locally it will not call remote and OnRemoteModuleDebugInfo
-  // are never called. For this case we need to make sure sampling report is
-  // updated here as well.
-  // TODO: LoadSymbolsFromModules should always call a callback even when
-  // symbols are loaded locally. Remove this call after this is done.
-  UpdateSamplingReport();
-  GOrbitApp->FireRefreshCallbacks();
+void OrbitApp::LoadRemoteModules(uint32_t process_id,
```

Comment:
Can we get rid of either this or `LoadModules` or is this called from somewhere else other than `LoadModules`?

Reply:
Or rather, is `LoadModules` necessary?

Reply:
I rewrote this so `LoadRemoteModules` is not necessary anymore and all the work is done in `LoadModules` (with a slightly different function signature), ptal